app/src/main/java/de/stephanlindauer/criticalmaps/adapter/ChatMessageAdapter.java

```diff
@@ -12,65 +12,77 @@
-        super(context, layoutResourceId, chatMessages);
+    public ChatMessageAdapter(ArrayList<IChatMessage> chatMessages) {
         this.chatMessages = chatMessages;
-        this.context = context;
     }
 
-    @Override
-    public View getView(int position, View convertView, ViewGroup parent) {
-        LayoutInflater inflater = (LayoutInflater) context
-                .getSystemService(Context.LAYOUT_INFLATER_SERVICE);
+    public class OutgoingChatMessageViewHolder extends RecyclerView.ViewHolder {
 
-        IChatMessage currentMessage = chatMessages.get(position);
+        @Bind(R.id.firstLine)
+        TextView labelView;
 
-        if (currentMessage instanceof ReceivedChatMessage)
-            return buildReceivedMessageView((ReceivedChatMessage) currentMessage, inflater, parent);
-        else if (currentMessage instanceof OutgoingChatMessage)
-            return buildOutgoingMessageView((OutgoingChatMessage) currentMessage, inflater, parent);
+        @Bind(R.id.secondLine)
+        TextView valueView;
 
-        return null;
-    }
+        private final DateFormat dateFormatter = DateFormat.getDateTimeInstance(DateFormat.DEFAULT, DateFormat.SHORT, Locale.getDefault());
+        private final Context context;
 
-    private View buildOutgoingMessageView(OutgoingChatMessage currentMessage, LayoutInflater inflater, ViewGroup parent) {
+        public OutgoingChatMessageViewHolder(View itemView) {
+            super(itemView);
+            context = itemView.getContext();
+            ButterKnife.bind(this, itemView);
+        }
 
-        View rowView = inflater.inflate(R.layout.view_outgoing_chatmessage, parent, false);
-
-        TextView labelView = (TextView) rowView.findViewById(R.id.firstLine);
-        TextView valueView = (TextView) rowView.findViewById(R.id.secondLine);
-
-        labelView.setText(R.string.chat_sending);
-        valueView.setText(currentMessage.getMessage());
-
-        return rowView;
+        public void bind(IChatMessage message) {
+            valueView.setText(message.getMessage());
+            if (message instanceof ReceivedChatMessage) {
+                dateFormatter.setTimeZone(TimeZone.getDefault());
+                labelView.setText(TimeToWordStringConverter.getTimeAgo(((ReceivedChatMessage) message).getTimestamp(), context));
+            } else {
+                labelView.setText(R.string.chat_sending);
+            }
+        }
     }
 
-    private View buildReceivedMessageView(ReceivedChatMessage currentMessage, LayoutInflater inflater, ViewGroup parent) {
-
-        View rowView = inflater.inflate(R.layout.view_chatmessage, parent, false);
-
-        TextView labelView = (TextView) rowView.findViewById(R.id.firstLine);
-        TextView valueView = (TextView) rowView.findViewById(R.id.secondLine);
+    @Override
+    public OutgoingChatMessageViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
```

Comment:
@ligi Are you actually abusing this `Viewholder` to update content for both `ReceivedChatMessage` and `OutgoingChatMessage`? If so I would suggest renaming it to `ChatMessageViewHolder` to avoid confusion. And, I would personally put the `ViewHolder` in a separate file for readability reasons.